libfreerdp/codec/dsp.c

```diff
@@ -238,9 +240,14 @@ static BOOL freerdp_dsp_resample(FREERDP_DSP_CONTEXT* context, const BYTE* src,
 	/* We want to ignore differences of source and destination format. */
 	format = *srcFormat;
 	format.wFormatTag = WAVE_FORMAT_UNKNOWN;
+	format.wBitsPerSample = 0;
```

Comment:
Maybe just initialize like this: `AUDIO_FORMAT format = { 0 };`

Reply:
If we go deeper in details, the audio_format_compatible checks four things:
* the format tag, but we ignore it here
* the bits per sample, now the same
* the number of channel, but it's always the same because of line 1166 (later instruction just before the call to freerdp_dsp_resample).
* the number of sample per seconds, the only one, finally

So finally, we could directly replace the audio_format_compatible call by comparing only the number of sample per seconds.

But If I'm wrong, it means that the audio_format_compatible still need to compare the number of channel, and so you ask me to replace:

```
	format = *srcFormat;
	format.wFormatTag = WAVE_FORMAT_UNKNOWN;
	format.wBitsPerSample = 0;
```
by
```
	AUDIO_FORMAT format = { 0 };
	[...]
	format.nChannels = srcFormat->nChannels;
	format.nSamplesPerSec = srcFormat->nSamplePerSec;
```

What do you choose?

Reply:
@Pollux42 looks like I overlooked `format = *srcFormat;` here, so the original is actually quite fine. (Just reset what you want to ignore in the comparison), sorry for the noise.